lib/Frontend/Frontend.cpp

```diff
@@ -74,17 +74,51 @@ void CompilerInstance::setPrimarySourceFile(SourceFile *SF) {
 bool CompilerInstance::setup(const CompilerInvocation &Invok) {
   Invocation = Invok;
 
+  setUpLLVMArguments();
+  setUpDiagnosticOptions();
+
+  // If we are asked to emit a module documentation file, configure lexing and
+  // parsing to remember comments.
+  if (!Invocation.getFrontendOptions().ModuleDocOutputPath.empty())
+    Invocation.getLangOptions().AttachCommentsToDecls = true;
+
+  // If we are doing index-while-building, configure lexing and parsing to
+  // remember comments.
```

Comment:
Seems like these should be collapsed into the same conditional.

Reply:
Left alone, as agreed.